### PR TITLE
Add new variations and sizes of progress bar

### PR DESCRIPTION
**Pull request by LukasHirt**
We've added new variations and sizes of the progress bar component. Variations change the color of the progress bar and can be either `primary` or `warning`. Sizes affect the height of the progress bar where it can be either `default` or `small`.

### CODE REVIEW

**Comment by micbar** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 34
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039&bid=19398742)

File: src/styles/theme/oc-progress.scss [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+      transition: width $transition-duration-short ease;
+      // Remove right border in IE11 and Edge
+      border: 0;
+    }
+
+    &-small {
+      height: $progress-height-small;
+    }
+
+    &-warning {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:62]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+    &:focus {
+      outline: none;
+    }
+
+    // Remove animated circles for indeterminate state in IE11 and Edge
+    &:indeterminate {
+      color: transparent;
+    }
+
+    &::-webkit-progress-bar {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:33]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    display: block;
+    width: 100%;
+    // Remove default style
+    border: 0;
+    // Set background color for progress container in Firefox, IE11 and Edge
+    background-color: $global-inverse-color;
+    height: $progress-height;
+
+    &:focus {
+      outline: none;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:25]: Codacy found an issue: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+// Component
+// ========================================================================
+
+.oc-progress {
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
+    vertical-align: baseline;
+    // Remove default style
+    -webkit-appearance: none;
+    -moz-appearance: none;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:15]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/_owncloud.scss [hunk 5]
```diff
@@ -49,6 +49,7 @@
 @import "theme/oc-spinner.scss";
 @import "theme/oc-modal.scss";
 @import "theme/oc-sidebar.scss";
+@import "theme/oc-progress";
```
Comment (micbar) [inline src/styles/_owncloud.scss:52]: Codacy found an issue: [Unexpected duplicate @import rule theme/oc-progress (no-duplicate-at-import-rules)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+$progress-height-small: 5px !default;
+
+// Component
+// ========================================================================
+
+.oc-progress {
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
+    vertical-align: baseline;
+    // Remove default style
+    -webkit-appearance: none;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:14]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+    &::-webkit-progress-value {
+      background-color: $action-primary;
+      transition: width $transition-duration-short ease;
+    }
+
+    &::-moz-progress-bar {
+      background-color: $action-primary;
+    }
+
+    &::-ms-fill {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:51]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+$progress-height: 15px !default;
+$progress-height-small: 5px !default;
+
+// Component
+// ========================================================================
+
+.oc-progress {
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
+    vertical-align: baseline;
+    // Remove default style
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:13]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+    background-color: $global-inverse-color;
+    height: $progress-height;
+
+    &:focus {
+      outline: none;
+    }
+
+    // Remove animated circles for indeterminate state in IE11 and Edge
+    &:indeterminate {
+      color: transparent;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:30]: Codacy found an issue: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+// ========================================================================
+
+$progress-height: 15px !default;
+$progress-height-small: 5px !default;
+
+// Component
+// ========================================================================
+
+.oc-progress {
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:11]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    vertical-align: baseline;
+    // Remove default style
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    display: block;
+    width: 100%;
+    // Remove default style
+    border: 0;
+    // Set background color for progress container in Firefox, IE11 and Edge
+    background-color: $global-inverse-color;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:21]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+.oc-progress {
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
+    vertical-align: baseline;
+    // Remove default style
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    display: block;
+    width: 100%;
+    // Remove default style
+    border: 0;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:19]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+    &:indeterminate {
+      color: transparent;
+    }
+
+    &::-webkit-progress-bar {
+      background-color: $global-inverse-color;
+    }
+
+    // Remove progress bar for indeterminate state in Firefox
+    &:indeterminate::-moz-progress-bar {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:38]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+      width: 0;
+    }
+
+    &::-webkit-progress-value {
+      background-color: $action-primary;
+      transition: width $transition-duration-short ease;
+    }
+
+    &::-moz-progress-bar {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:47]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 15]
```diff
@@ -0,0 +1,75 @@
...
+    &::-webkit-progress-bar {
+      background-color: $global-inverse-color;
+    }
+
+    // Remove progress bar for indeterminate state in Firefox
+    &:indeterminate::-moz-progress-bar {
+      width: 0;
+    }
+
+    &::-webkit-progress-value {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:42]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 16]
```diff
@@ -0,0 +1,75 @@
...
+      background-color: $action-primary;
+      transition: width $transition-duration-short ease;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:44]: Codacy found an issue: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 17]
```diff
@@ -0,0 +1,75 @@
...
+    -moz-appearance: none;
+    display: block;
+    width: 100%;
+    // Remove default style
+    border: 0;
+    // Set background color for progress container in Firefox, IE11 and Edge
+    background-color: $global-inverse-color;
+    height: $progress-height;
+
+    &:focus {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:24]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 18]
```diff
@@ -0,0 +1,75 @@
...
+      outline: none;
+    }
+
+    // Remove animated circles for indeterminate state in IE11 and Edge
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:28]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 19]
```diff
@@ -0,0 +1,75 @@
...
+      background-color: $global-inverse-color;
+    }
+
+    // Remove progress bar for indeterminate state in Firefox
+    &:indeterminate::-moz-progress-bar {
+      width: 0;
+    }
+
+    &::-webkit-progress-value {
+      background-color: $action-primary;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:43]: Codacy found an issue: [Expected indentation of 4 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 20]
```diff
@@ -0,0 +1,75 @@
...
+
+$progress-height: 15px !default;
+$progress-height-small: 5px !default;
+
+// Component
+// ========================================================================
+
+.oc-progress {
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
+    vertical-align: baseline;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:12]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 21]
```diff
@@ -0,0 +1,75 @@
...
+    // Set background color for progress container in Firefox, IE11 and Edge
+    background-color: $global-inverse-color;
+    height: $progress-height;
+
+    &:focus {
+      outline: none;
+    }
+
+    // Remove animated circles for indeterminate state in IE11 and Edge
+    &:indeterminate {
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:29]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 22]
```diff
@@ -0,0 +1,75 @@
...
+    width: 100%;
+    // Remove default style
+    border: 0;
+    // Set background color for progress container in Firefox, IE11 and Edge
+    background-color: $global-inverse-color;
+    height: $progress-height;
+
+    &:focus {
+      outline: none;
+    }
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:26]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss (hunk 14 above)
Comment (micbar) [inline src/styles/theme/oc-progress.scss:47]: Codacy found an issue: [Expected selector ".oc-progress::-moz-progress-bar" to come before selector ".oc-progress:indeterminate::-moz-progress-bar" (no-descending-specificity)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 23]
```diff
@@ -0,0 +1,75 @@
...
+    // Add the correct vertical alignment in Chrome, Firefox, and Opera.
+    vertical-align: baseline;
+    // Remove default style
+    -webkit-appearance: none;
+    -moz-appearance: none;
+    display: block;
+    width: 100%;
+    // Remove default style
+    border: 0;
+    // Set background color for progress container in Firefox, IE11 and Edge
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:20]: Codacy found an issue: [Expected indentation of 2 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/styles/theme/oc-progress.scss [hunk 24]
```diff
@@ -0,0 +1,75 @@
...
+      border: 0;
+    }
+
+    &-small {
+      height: $progress-height-small;
+    }
+
+    &-warning {
+      &::-webkit-progress-value {
+        background-color: $warning-background;
```
Comment (micbar) [inline src/styles/theme/oc-progress.scss:64]: Codacy found an issue: [Expected indentation of 6 spaces (indentation)](https://app.codacy.com/gh/owncloud/owncloud-design-system/pullRequest?prid=5920039)

File: src/elements/OcProgress.vue [hunk 25]
```diff
@@ -35,11 +32,40 @@ export default {
...
     },
+    /**
+     * The size of the progress bar.
+     * Can be `default` or `small`
+     */
+    size: {
+      type: String,
+      required: false,
+      default: "default"
+    },
```
Comment (kulmann) [inline src/elements/OcProgress.vue:43]: please add validation for possible sizes

File: src/elements/OcProgress.vue [hunk 26]
```diff
@@ -35,11 +32,40 @@ export default {
...
+    /**
+     * The variation of the progress bar.
+     * Can be `primary` or `warning`
+     */
+    variation: {
+      type: String,
+      required: false,
+      default: "primary"
```
Comment (kulmann) [inline src/elements/OcProgress.vue:51]: please add validation for allowed variations

File: src/elements/OcProgress.vue (hunk 25 above)
Comment (LukasHirt) [reply src/elements/OcProgress.vue:46]: Done :wink:

File: src/elements/OcProgress.vue (hunk 26 above)
Comment (LukasHirt) [reply src/elements/OcProgress.vue]: Done :wink:

**Pull request merged**